pkg/brownfield/listeners.go

```diff
@@ -6,11 +6,87 @@
 package brownfield
 
 import (
+	"strings"
+
 	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/golang/glog"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/utils"
 )
 
 type listenersByName map[listenerName]n.ApplicationGatewayHTTPListener
 
+// GetBlacklistedListeners filters the given list of health probes to the list Probes that AGIC is allowed to manage.
+func (er ExistingResources) GetBlacklistedListeners() ([]n.ApplicationGatewayHTTPListener, []n.ApplicationGatewayHTTPListener) {
+	blacklistedListenersSet := er.getBlacklistedListenersSet()
+	var blacklisted, nonBlacklisted []n.ApplicationGatewayHTTPListener
+	for _, listener := range er.Listeners {
+		listenerNm := listenerName(*listener.Name)
+		if _, exists := blacklistedListenersSet[listenerNm]; exists {
+			glog.V(5).Infof("[brownfield] Listener %s is blacklisted", listenerNm)
+			blacklisted = append(blacklisted, listener)
+			continue
+		}
+		glog.V(5).Infof("[brownfield] Listener %s is not blacklisted", listenerNm)
+		nonBlacklisted = append(nonBlacklisted, listener)
+	}
+	return blacklisted, nonBlacklisted
+}
+
+// MergeListeners merges list of lists of listeners into a single list, maintaining uniqueness.
+func MergeListeners(listenerBuckets ...[]n.ApplicationGatewayHTTPListener) []n.ApplicationGatewayHTTPListener {
+	uniq := make(listenersByName)
+	for _, bucket := range listenerBuckets {
+		for _, listener := range bucket {
+			uniq[listenerName(*listener.Name)] = listener
+		}
+	}
+	var merged []n.ApplicationGatewayHTTPListener
+	for _, listener := range uniq {
+		merged = append(merged, listener)
```

Comment:
we have to make sure that listeners don't have same hostname/port. If yes, then we have to delete one.